roles/reproducer/templates/content-provider.yml.j2

```
@@ -54,54 +54,142 @@
-    # Ensure we get consistent name for the operator, linked to
-    # https://github.com/openstack-k8s-operators/ci-framework/pull/621
-    - name: Set var for cifmw_operator_build_operators var
+        - name: Output needed content into consumable environment
+          ansible.builtin.copy:
+            dest: "{{ ansible_user_dir }}/{{ job_id }}-params/content-provider.yml"
+            content: |-
+              {{
+                {'cifmw_operator_build_output': cifmw_operator_build_output,
+                 'content_provider_registry_ip': cifmw_rp_registry_ip
+                } | to_nice_yaml
+              }}
+    {% endraw %}
+
+    - name: Run openstack content provider
       when:
-        - zuul is defined
-        - "'project' in zuul"
-        - "'short_name' in zuul.project"
-      ansible.builtin.set_fact:
-        cifmw_operator_build_operators:
-          - name: "openstack-operator"
-            src: "~/src/github.com/{{ cifmw_operator_build_org }}/openstack-operator"
-            image_base: >-
-              {{ zuul.project.short_name | split('-') | reject('search','operator') | join('-') }}
-
-    - name: Build Operators
-      ansible.builtin.include_role:
-        name: operator_build
-
-    - name: Get the containers list from container registry
-      ansible.builtin.uri:
-        url: "http://{{ cifmw_rp_registry_ip }}:5001/v2/_catalog"
-        return_content: true
-      register: cp_imgs
-
-    - name: Add the container list to file
-      ansible.builtin.copy:
-        content: "{{ cp_imgs.content }}"
-        dest: "{{ ansible_user_dir }}/local_registry.log"
-        mode: "0644"
-
-    - name: Output needed content into consumable environment
-      ansible.builtin.copy:
-        dest: "{{ ansible_user_dir }}/{{ job_id }}-params/content-provider.yml"
-        content: |-
-          {{
-            {'cifmw_operator_build_output': cifmw_operator_build_output,
-             'content_provider_registry_ip': cifmw_rp_registry_ip
-            } | to_nice_yaml
-          }}
-{% endraw %}
+        - openstack_content_provider | default(false) | bool
+        - not operator_content_provider | default(false) | bool
+      block:
+        - name: "Run ci/playbooks/tcib/tcib.yml"
+          hosts: "{{ cifmw_target_host | default('localhost') }}"
+          gather_facts: true
+          tasks:
+            - name: Filter out host if needed
+              when:
+                - cifmw_zuul_target_host is defined
+                - cifmw_zuul_target_host != 'all'
+                - inventory_hostname != cifmw_zuul_target_host
+              ansible.builtin.meta: end_host
+
+        - name: Discover the host ip
+          ansible.builtin.set_fact:
+            node_ip: >-
+              {{ hostvars[groups.all[0]].ansible_host if hostvars[groups.all[0]].ansible_host
+              is match("[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+")
+              else hostvars[inventory_hostname]['ansible_default_ipv4']['address'] }}
```

This part is not needed. We can replace node_ip to cifmw_rp_registry_ip. or we can modify this task.
```suggestion
        - name: Discover the host ip
          ansible.builtin.set_fact:
            node_ip: "{{ cifmw_rp_registry_ip }} "
```

---

ack, done